main.cpp

```diff
@@ -122,9 +124,37 @@ int main(int argc, char const* argv[]) {
     // load triangles from the scene into a kd-tree
     std::cerr << "Loading triangles and building kd-tree..." << std::endl;
     Runtime loading_time;
-    auto triangles = triangles_from_scene(scene);
-    Stats::instance().num_triangles = triangles.size();
-    KDTree tree(std::move(triangles));
+
+    // Load KDTree from cache if it exists or build it.
+    size_t kdtree_runtime_ms = 0;
+    KDTree tree;
+    {
+        Runtime runtime(kdtree_runtime_ms);
+        std::ifstream kdtree_cache("kdtree.cache");
+        if(kdtree_cache.is_open()) {
+            {
+                cereal::PortableBinaryInputArchive iarchive(kdtree_cache);
+                iarchive(tree);
+            }
+            kdtree_cache.close();
```

Comment:
No need to do it; it will be done automatically, when you leave the lexical scope.